controllers/host/host_controller.go

```diff
@@ -1616,6 +1617,79 @@ func (r *HostReconciler) ReconcileDeletedHost(client *gophercloud.ServiceClient,
 	return nil
 }
 
+// TriggerNetworkReconcilerAndUnlock is used to trigger network reconciliation after making changes to platform network.
+// After successful reconciliation unlock-strategy is applied if the system is in locked state.
+func (r *HostReconciler) TriggerNetworkReconcilerAndUnlock(client *gophercloud.ServiceClient, instance *starlingxv1.Host) (err error) {
```

Comment:
The host_controller already do a reconcile when the host is locked disabled. Is there any special case that we need to trigger this again?

Reply:
In order to trigger reconciliation from host controller and use VIM strategy to unlock, we have to set the deploymentScope to `principal` - I think it's not good idea to update deploymentScope without getting it from the user.

Just updating `instance.Status.Reconciled` to false and letting the `HostReconciler` handle reconciliation would mean, the `HostReconciler` would use Day-1 procedure, ie. direct API call to sysinv and not use VIM strategy to unlock the system.

Given the use case, we are fixing interface-network association from outisde of `Reconcile` body.